mlflow/cli.py

```diff
@@ -554,6 +562,16 @@ def gc(older_than, backend_store_uri, run_ids):
     else:
         run_ids = run_ids.split(",")
 
+    mlflow_client = mlflow.MlflowClient()
+    deleted_experiments = [
+        experiment
+        for experiment in mlflow_client.search_experiments(view_type=ViewType.DELETED_ONLY)
+        if experiment.last_update_time <= time_delta
```

Comment:
`last_update_time` was added in MLflow 1.29.0. It's `None` for experiments logged before that version. `None <= int_num` results in `TypeError`.

Reply:
```suggestion
        if (experiment.last_update_time or float('inf')) <= time_delta
```

We might need to fallback to `float('inf')`.

Reply:
```suggestion
        if experiment.last_update_time is not None or experiment.last_update_time <= time_delta
```

Reply:
Can I ignore `if experiment.last_update_time <= time_delta` condition for older versions?

Reply:
I think the expected behavior is:

- If `--older-than` is unspecified, we don't need to look at `last_update_time`. 
- If it's specified, we need to exclude experiments with `last_update_time=None` because we aren't sure when they were created. Some of them might have been created after the specified time threshold.